### DISCUSSION TITLE
How to let CloudFormation Guard ignore lines starting with !Ref when I am searching for !Sub

### DISCUSSION THREAD

**Discussion by Aaron-Garrett** [community/community#114274] [Programming Help]
### Body

My code looks like this:

### Defining the roles
let aws_iam_entities_no_managed_policy = Resources.*[
  Type in [ /AWS::IAM::User/,
            /AWS::IAM::Role/,
            /AWS::IAM::Group/ ]
  Metadata.guard.SuppressedRules not exists or
  Metadata.guard.SuppressedRules.* != "IAM_ONLY_ALLOW_NONAWSManagedPolicies"
]

### Defining the rule that everything will compare against and making sure that the role has managed policies attached
rule IAM_ONLY_ALLOW_NONAWSManagedPolicies when %aws_iam_entities_no_managed_policy !empty {
  when %aws_iam_entities_no_managed_policy.Properties.ManagedPolicyArns !empty {

### Defining the expectations for the managed policies (It must NOT start with AWS as those are aws-managed policies and typically have too broad of permissions)
    %aws_iam_entities_no_managed_policy.Properties.ManagedPolicyArns[*]."Fn::Sub" !in [/^arn:aws:iam::\${AWS::AccountId}:policy\/AWS/,"arn:aws:iam::${AWS::AccountId}:policy/VMImportExportRoleForAWSConnector"]

### Violation message
    <<
      Violation: Only select managed policies are allowed on IAM Users, Roles, or Groups.
      Fix: Remove the disallowed policies from any IAM Users, Roles, or Groups.
    >>
  }
}

# What I am looking for
But this code is flagging lines that start with !Ref instead of !Sub instead of just skipping them. I do not care if they are referencing something else in their code, so how can I change this code to ignore, or allow, anything starting with the Ref function and continue to flag any subs for Managed Policies that begin with AWS?

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by Illia65** [comment]
Response#1
